toolbox/atmospheric_tools/grib2_to_fv3_regrid.py

```diff
@@ -0,0 +1,609 @@
+#!/usr/bin/env python3
+"""
+GRIB2 to FV3 Regridding Tool
+
+This script reads GRIB2 aerosol data (e.g., from GEFS chemistry products) and
+regrids it both horizontally and vertically to FV3 cube sphere tiles.
+
+Author: SCHWARTZ Toolbox
+Date: 2025
+"""
+
+import argparse
+import os
+import sys
+from pathlib import Path
+import logging
+
+try:
+    import numpy as np
+    import xarray as xr
+    import pygrib
+    from scipy.interpolate import RegularGridInterpolator, interp1d
+    from scipy.spatial.distance import cdist
+    DEPENDENCIES_AVAILABLE = True
+except ImportError as e:
+    print(f"Note: Scientific computing dependencies not available: {e}")
+    print("For full functionality, install dependencies using:")
+    print("  pip install -r requirements.txt")
+    print("\nArgument parsing and basic functionality still available.")
+    DEPENDENCIES_AVAILABLE = False
+    
+    # Define dummy classes for when dependencies are not available
+    class RegularGridInterpolator:
+        def __init__(self, *args, **kwargs):
+            pass
+    
+    class interp1d:
+        def __init__(self, *args, **kwargs):
+            pass
+
+
+class FV3CubeSphere:
+    """Class to handle FV3 cube sphere grid operations."""
+    
+    def __init__(self, npx=384, orography_file=None):
+        """
+        Initialize FV3 cube sphere grid.
+        
+        Parameters:
+        -----------
+        npx : int
+            Number of grid points in each direction (default: 384 for C384)
+        orography_file : str, optional
+            Path to orography file containing geolon/geolat coordinates
+        """
+        self.npx = npx
+        self.npy = npx
+        self.ntiles = 6
+        self.orography_file = orography_file
+        
+    def generate_coordinates(self):
+        """
+        Generate cube sphere coordinates for all 6 tiles.
+        
+        Returns:
+        --------
+        dict : Dictionary containing lat/lon coordinates for each tile
+        """
+        coords = {}
+        
+        # If orography file is provided, load coordinates from it
+        if self.orography_file and os.path.exists(self.orography_file):
```

Comment:
Each tile will have its own orography file, the user should provide a prefix and ends in tileN.nc where N is the tile number